http/src/test/java/com/redhat/lightblue/client/http/LightblueHttpClientTest.java

```diff
@@ -22,29 +26,93 @@ public void testPojoMapping() throws IOException {
-                .equals(results[0]));
+        Assert.assertTrue(new SimpleModelObject("idhash", "value").equals(results[0]));
+    }
+
+    @Test
+    public void testPojoMappingWithErrorInResponse() throws IOException {
+        expectedEx.expect(RuntimeException.class);
+        expectedEx.expectMessage("Error in lightblue response");
+
+        DataFindRequest findRequest = new DataFindRequest("foo", "bar");
+
+        findRequest.where(withValue("foo = bar"));
+        findRequest.select(includeField("_id"));
+
+        String response = "{\"context\": \"rest/FindCommand/esbMessage\", \"errorCode\": \"rest-crud:RestFindError\", \"msg\": \"java.lang.IllegalArgumentException: Cannot call method public static com.redhat.lightblue.crud.FindRequest com.redhat.lightblue.crud.FindRequest.fromJson(com.fasterxml.jackson.databind.node.ObjectNode)\", \"objectType\": \"error\"}";
+
+        client.setLightblueResponse(response);
+
+        client.data(findRequest, SimpleModelObject[].class);
     }
 
-    @Test(expected = LightblueHttpClientException.class)
+    /**
+     * This is bad json data in order to test a scenario.
+     */
+    @Test
     public void testPojoMappingWithParsingError() throws IOException {
+        expectedEx.expect(RuntimeException.class);
+        expectedEx.expectMessage("Error parsing lightblue response");
+
         DataFindRequest findRequest = new DataFindRequest("foo", "bar");
 
         findRequest.where(withValue("foo = bar"));
         findRequest.select(includeField("_id"));
 
-        String response = "{\"processed\":\"<p>This is not json</p>\"}";
+        String response = "{\"processed\": \"bad json\"}";
 
         client.setLightblueResponse(response);
 
         client.data(findRequest, SimpleModelObject[].class);
     }
 
-}
+    @Test
+    public void testPojoMappingWithEmptyProcessedResults() throws IOException {
+        DataFindRequest findRequest = new DataFindRequest("foo", "bar");
+
+        findRequest.where(withValue("foo = bar"));
+        findRequest.select(includeField("_id"));
+
+        client.setLightblueResponse("{\"matchCount\": 0, \"modifiedCount\": 0}");
+
+        SimpleModelObject[] results = client.data(findRequest, SimpleModelObject[].class);
+
+        Assert.assertNull(results);
+    }
+
+    @Test
+    public void testPojoMappingWithEmptyProcessedResults_EmptyArrayNode() throws IOException {
+        DataFindRequest findRequest = new DataFindRequest("foo", "bar");
+
+        findRequest.where(withValue("foo = bar"));
+        findRequest.select(includeField("_id"));
+
+        client.setLightblueResponse("{\"status\":\"COMPLETE\",\"modifiedCount\":0,\"matchCount\":0,\"processed\":[]}");
+
+        SimpleModelObject[] results = client.data(findRequest, SimpleModelObject[].class);
+
+        Assert.assertEquals(0, results.length);
+    }
+
+    @Test
+    public void testPojoMappingWithEmptyProcessedResults_ForSingleResult() throws IOException {
```

Comment:
testPojoMappingWithEmptyProcessedResults_ForSingleResult name is misleading, maybe call it testPojoMappingWithEmptyProcessedResults_ForEmptyJsonNode ?  I see in the test it's verifying the client gracefully handles parsing an empty array to a single null result.